app/src/main/java/com/realityexpander/tasky/agenda_feature/presentation/event_screen/components/PhotoModal.kt

```diff
@@ -0,0 +1,181 @@
+
+    Box(
+        modifier = Modifier
+            .fillMaxSize()
+            .background(Color.Black.copy(alpha = .5f))
+    ) {
+
+        Column(
+            modifier = Modifier
+                .fillMaxSize()
+                .background(MaterialTheme.colors.surface)
+
+        ) {
+
+            Row(
+                horizontalArrangement = Arrangement.SpaceBetween,
+                verticalAlignment = Alignment.CenterVertically,
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .padding(DP.small)
+            ) {
+
+                // Close button
+                Icon(
+                    Icons.Outlined.Close,
+                    tint = MaterialTheme.colors.onSurface,
+                    contentDescription = stringResource(R.string.back_description),
+                    modifier = Modifier
+                        .size(32.dp)
+                        .align(Alignment.CenterVertically)
+                        .clickable { onCancel() }
+                )
+
+                // Title
+                Text(
+                    title,
+                    color = MaterialTheme.colors.onSurface,
+                    style = MaterialTheme.typography.h4,
+                    textAlign = TextAlign.Start,
+                    modifier = Modifier
+                )
+
+                // Remove button
+                Icon(
+                    Icons.Outlined.Delete,
+                    tint = if (isRemoveEnabled) MaterialTheme.colors.onSurface else Color.Transparent,
+                    contentDescription = stringResource(R.string.remove_description),
+                    modifier = Modifier
+                        .size(32.dp)
+                        .align(Alignment.CenterVertically)
+                        .clickable(isRemoveEnabled) {
+                            isConfirmRemoveDialogVisible = true
+                        }
+                )
+            }
+
+            Divider()
+
+            // Photo
+            when (photo) {
+                is Photo.Local -> {
+                    AsyncImage(
+                        model = photo.uri,
+                        contentDescription = stringResource(id = R.string.event_description_photo),
+                        modifier = Modifier
+                            .fillMaxSize()
+                            .weight(1f),
+                        contentScale = ContentScale.Fit
+                    )
+                }
+                is Photo.Remote -> {
+                    AsyncImage(
+                        model = photo.url,
+                        contentDescription = stringResource(id = R.string.event_description_photo),
+                        modifier = Modifier
+                            .fillMaxSize()
+                            .weight(1f),
+                        contentScale = ContentScale.Fit
+                    )
+                }
+            }
+        }
```

Comment:
Both these `AsyncImages` seem to be exactly the same except for the model. So here it would make more sense to instead use the `when` just for the model to get rid of needing 2 composables

Reply:
OK will swap out for when in the model.